server/transcription_api.go

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) 2022-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package main
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/mattermost/calls-offloader/public/job"
+
+	"github.com/mattermost/mattermost/server/public/model"
+)
+
+const transcriptionJobStartTimeout = 30 * time.Second
+
+func (p *Plugin) transcriptionJobTimeoutChecker(callID, jobID string) {
+	time.Sleep(transcriptionJobStartTimeout)
+
+	state, err := p.lockCall(callID)
+	if err != nil {
+		p.LogError("failed to lock call", "err", err.Error())
+		return
+	}
+	defer p.unlockCall(callID)
+
+	trState, err := state.getTranscription()
+	if err != nil {
+		p.LogError("failed to get transcription state", "error", err.Error())
+		return
+	}
+
+	// If the transcription hasn't started (bot hasn't joined yet) we notify the
+	// client.
+	if trState.StartAt == 0 {
+		if trState.JobID != jobID {
+			p.LogInfo("a new job has started in between, exiting", "callID", callID, "jobID", jobID)
+			return
+		}
+
+		p.LogError("timed out waiting for transcriber bot to join", "callID", callID, "jobID", jobID)
+
+		state.Call.Transcription = nil
+		if err := p.kvSetChannelState(callID, state); err != nil {
+			p.LogError("failed to set channel state", "err", err.Error())
+			return
+		}
+
+		clientState := trState.getClientState()
+		clientState.Err = "failed to start transcriber job: timed out waiting for bot to join call"
+		clientState.EndAt = time.Now().UnixMilli()
+
+		if state.Call.Recording != nil && state.Call.Recording.EndAt == 0 {
+			if _, _, err := p.stopRecordingJob(state, callID); err != nil {
+				p.LogError("failed to stop recording job", "callID", callID, "err", err.Error())
+			}
+		}
```

Comment:
My understanding: we're stopping the recording if the transcriber bot fails to join, is that correct? I guess it keeps them linked... Maybe that is the best thing to do. (leaving this comment here just in case it sparks any ideas)